walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,64 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    ArrayList<TimeRange> validTimes = new ArrayList<TimeRange>();
+    Collection<String> wantedAttendees = request.getAttendees();
+    ArrayList<TimeRange> occupiedTimes = new ArrayList<TimeRange>();
+
+    TimeRange whole_day = TimeRange.WHOLE_DAY;
+
+    if(wantedAttendees.size() == 0) {
+        validTimes.add(whole_day);
+        return validTimes;
+    }
+
+    if(request.getDuration() > whole_day.duration()) {
+        return validTimes;
+    }
+
+    for(Event schedule : events) {
+        Collection<String> busyAttendees = schedule.getAttendees();
+
+        for(String person:wantedAttendees) {
+            if(busyAttendees.contains(person)) {
+                occupiedTimes.add(schedule.getWhen());
+            }
+        }
+    }
+
+    if(occupiedTimes.size()==0) {
+        validTimes.add(whole_day);
+        return validTimes;
+    }
+
+    Collections.sort(occupiedTimes, TimeRange.ORDER_BY_START);
+
+    TimeRange marker = TimeRange.START;
+
+    for(TimeRange meeting : occupiedTimes){
```

Comment:
the variable "meeting" is sort of misnamed (I expect this to be an event type object). Something like "meetingTime" or "existingMeetingTime" might be more appropriate.